File: services/ts-express-superstruct/spec_router.ts
import {Router} from 'express'
import {EchoService as EchoServiceV2} from './v2/echo_service'
import {echoRouter as echoRouterV2} from './v2/routing'
import {EchoService as EchoService} from './echo_service'
import {echoRouter as echoRouter} from './routing'
import {CheckService as CheckService} from './check_service'
import {checkRouter as checkRouter} from './routing'

export let specRouter = (echoServiceV2: EchoServiceV2, echoService: EchoService, checkService: CheckService) => {
    let router = Router()
    router.use('/v2', echoRouterV2(echoServiceV2))
    router.use('/', echoRouter(echoService))
    router.use('/', checkRouter(checkService))
    return router
}

File: services/scala-play/app/SpecRouter.scala
package app

import javax.inject._
import play.api.routing._

class SpecRouter @Inject()(echoV2: v2.routers.EchoRouter, echo: routers.EchoRouter, check: routers.CheckRouter) extends SimpleRouter {
  def routes: Router.Routes =
    Seq(
      echoV2.routes,
      echo.routes,
      check.routes,
    ).reduce { (r1, r2) => r1.orElse(r2) }
}
